rel/m_lib.c

```diff
@@ -1,371 +1,817 @@
+  f32 step;
+
+  if (now >= end) {
+    return 1.0f;
+  }
+  if (now <= start) {
+    return 0.0f;
+  }
+  total_delta = end - start;
+  now_delta = now - start;
+  if (total_delta < (accelerate + brake)) {
+    return 0.0f;
+  }
+
+  step = 1.0f / (((2.0f * total_delta) - accelerate) - brake);
+  if (accelerate != 0.0f) {
+    if (now_delta <= accelerate) {
+      percent = (now_delta * (step * now_delta));
+      percent /= accelerate;
+      return percent;
+    }
+    percent = step * accelerate;
+  } else {
+    percent = 0.0f;
+  }
+  if (now_delta <= (total_delta - brake)) {
+    percent += (step * 2.0f) * (now_delta - accelerate);
+    return percent;
+  }
+  percent += (2.0f * step * ((total_delta - accelerate) - brake));
+  if (brake != 0.0f) {
+    percent += step * brake;
+    if (now_delta < total_delta) {
+      f32 diff = total_delta - now_delta;
+      percent -= step * diff * diff / brake;
+    }
+  }
+  return percent;
+}
 
-    Matrix_mult(&play->unk_200C, 0);
-    Matrix_Position(src, dest);
-    w = play->unk_200C.ww + (
-          (play->unk_200C.wx * src->x) 
-        + (play->unk_200C.wy * src->y) 
-        + (play->unk_200C.wz * src->z));
-    dest->x = 160.0f + ((dest->x / w) * 160.0f);
-    dest->y = 120.0f - ((dest->y / w) * 120.0f);
+/**
+ * @brief Project a 3D world position onto a 2D screen position.
+ *
+ * @param play Pointer to a GAME_PLAY structure.
+ * @param wpos Pointer to the 3D world position (xyz_t).
+ * @param screen_pos Pointer to the resulting 2D screen position (xyz_t).
+ */
+extern void Game_play_Projection_Trans(GAME_PLAY* play, xyz_t* wpos,
+                                       xyz_t* screen_pos) {
+  f32 w;
+
+  Matrix_mult(&play->matrix, 0);
+  Matrix_Position(wpos, screen_pos);
+  w = play->matrix.ww +
+      ((play->matrix.wx * wpos->x) +
+       (play->matrix.wy * wpos->y) +
+       (play->matrix.wz * wpos->z));
+  screen_pos->x = (SCREEN_WIDTH_F / 2.0f) + ((screen_pos->x / w) * (SCREEN_WIDTH_F / 2.0f));
+  screen_pos->y = (SCREEN_HEIGHT_F / 2.0f) - ((screen_pos->y / w) * (SCREEN_HEIGHT_F / 2.0f));
 }
 
-f32 get_percent(s32 arg0, s32 arg1, s32 arg2) {
-    f32 temp_f0;
-    f32 var_f2;
-
-    var_f2 = 1.0f;
-    if (arg2 < arg1) {
-        var_f2 = 0.0f;
-    } else if (arg2 < arg0) {
-        temp_f0 = (f32) (arg0 - arg1);
-        if (temp_f0 != 0.0f) {
-            var_f2 = (f32) (arg2 - arg1) / temp_f0;
-            if (var_f2 > 1.0f) {
-                var_f2 = 1.0f;
-            }
-        }
+/**
+ * @brief Calculate the percentage of the input value within the specified range.
+ *
+ * @param max Maximum value for the range.
+ * @param min Minimum value for the range.
+ * @param x Input value to check.
+ * @return Percentage of the input value within the specified range.
+ */
+extern f32 get_percent(int max, int min, int x) {
+  f32 check;
+  f32 percent;
+
+  percent = 1.0f;
+  if (x < min) {
+    percent = 0.0f;
+  } else if (x < max) {
+    check = (f32)(max - min);
```

Comment:
The total_delta name from the accelbrake function's probably closer to what this is. Also no need to cast, that'll happen implicitly.